src/massive/munit/TestRunner.hx

```diff
@@ -182,29 +195,88 @@ class TestRunner implements IAsyncDelegateObserver
         suiteIndex = 0;
         clientCompleteCount = 0;
         Assert.assertionCount = 0; // don't really like this static but can't see way around it atm. ms 17/12/10
-        testSuites = [for(suiteType in testSuiteClasses) Type.createInstance(suiteType, emptyParams)];
+        emptyParams = new Array();
+        asyncDelegates = new Array<AsyncDelegate>();
+        testSuites = new Array<TestSuite>();
         startTime = Timer.stamp();
 
-        #if (neko || cpp || java)
-		var self = this;
-		var runThread:Thread = Thread.create(function()
-		{
-			self.execute();
-			while (self.running)
-			{
-				Sys.sleep(.2);
-			}
-			var mainThead:Thread = Thread.readMessage(true);
-			mainThead.sendMessage("done");
-		});
-		runThread.sendMessage(Thread.current());
-		Thread.readMessage(true);
+        for (suiteType in testSuiteClasses)
+        {
+            testSuites.push(Type.createInstance(suiteType, new Array()));
+        }
+
+        #if (!lime && !nme && (neko||cpp))
+            var self = this;
+            var runThread:Thread = Thread.create(function()
+            {
+                self.execute();
+                while (self.running)
+                {
+                    Sys.sleep(.2);
+                }
+                var mainThead:Thread = Thread.readMessage(true);
+                mainThead.sendMessage("done");
+            });
+
+            runThread.sendMessage(Thread.current());
+            Thread.readMessage(true);
         #else
         execute();
         #end
     }
 
-    function execute()
+    private function callHelperMethod ( method:Dynamic ):Void
+    {
+        try
+        {
+            /*
+                Wrapping in try/catch solves below problem:
+                    If @BeforeClass, @AfterClass, @Before, @After methods
+                    have any Assert calls that fail, and if they are not
+                    caught and handled here ... then TestRunner stalls.
+            */
+            Reflect.callMethod(activeHelper.test, method, emptyParams);
+        }
+        catch (e:Dynamic)
+        {
+            var testcaseData: Dynamic = activeHelper.current(); // fetch the test context
+            exceptionHandler ( e, testcaseData.result );
+        }
+    }
+
+
+    private inline function exceptionHandler ( e:Dynamic, result:TestResult ):Void
+    {
+        #if hamcrest
+        if (Std.is(e, org.hamcrest.AssertionException))
+        {
+            e = new AssertionException(e.message, e.info);
+        }
+        #end
+
+        result.executionTime = Timer.stamp() - testStartTime;
+
+        if (Std.is(e, AssertionException))
+        {
+            result.failure = e;
+            failCount++;
+            for (c in clients)
+                c.addFail(result);
+        }
+        else
+        {
+            if (!Std.is(e, MUnitException))
+                e = new UnhandledException(e, result.location);
```

Comment:
268-269 but 
```haxe
 if (Std.is(e, org.hamcrest.AssertionException))
 {
            e = new AssertionException(e.message, e.info);
 }
```
;(

Reply:
This seems similar to the current sources:

https://github.com/massiveinteractive/MassiveUnit/blob/2eea63ec50751bd3d3f0b65b8c068ee22e6e5433/src/massive/munit/TestRunner.hx#L310-L327